unixfs/pb/unixfs.proto

```diff
@@ -1,24 +1,79 @@
 package unixfs.pb;
 
+/* FilePart represents a Unix file. */
+//
+// TODO: Rename. What name to use: `File`, `FilePart`, `FileChunk`?
+// `FilePart` makes it easier to understand the recursive nature of
+// this data structure, but the `Part` name throughout the code
+// base could confuse the reader thinking he's holding only a part
+// of the file when most of the time he actually has the root
+// representing the entire file.
+//
+// TODO: How much importance to give to the underlying DAG layer.
+// I like the File/FilePart/Fractal idea, it doesn't need to be backed
+// up by a DAG, but it does (and probably always will) so it may be
+// easier to describe it as part of the DAG (coupling both ideas).
+// The more I write docs in this file the more references I'm making
+// to nodes (i.e., DAG). It boils down to: is it easier to reason
+// about UnixFS with or without DAG (or something in between, e.g.,
+// using minor references as examples).
+//
+// TODO: Clearly indicate that we are talking about *any* file type (Unix style).
+// Files can be regular files, directories (what else?). That logic is being
+// disrupted with types like `Metadata` and `HAMTShard` which make me
+// rethink what is actually a file in the UnixFS specification.
+//
+// TODO: Following a question in `filesize`, the leaf and internal nodes
+// need to be differentiated, a leaf has no `blocksizes` and a internal
+// has no `Data` (is this correct?), there should even be different message
+// types for that (not happening, that will definitely break API).
 message Data {
+    // Type of file.
```

Comment:
I wouldnt say this is the type of the 'file'. Its a type for this data node. A file is a more abstract concept than this type

Reply:
Thanks! This comment was an eye-opener, those types are very misleading and there is an *implicit* "more abstract concept" on top of them that should be clarified, https://github.com/ipfs/go-ipfs/issues/5157.